axlearn/common/trainer.py

```diff
@@ -847,6 +850,7 @@ def _prepare_training(self, prng_key: Tensor) -> bool:
             with fs.open(os.path.join(cfg.dir, "model_analysis.txt"), "w") as f:
                 f.write(model_analysis)
 
+        self._maybe_record_event(measurement.Event.END_TRAINING_PREPARATION)
```

Comment:
What's usually considered part of "training preparation"? Should we count the jit compilation below as a potentially substantial part of it? What about the checkpoint restoration above?

Reply:
This would include the time spent on the creation of checkpoint managers, checkpoint loading, running mesh and model optimizers etc.

The JIT compilation is currently computed algorithmically based on the entire timeline of events (other recorded logs) and is included in the "program startup" badput that is meant to measure the time spent on framework specific function transformations (such as JAX tracing), compilation tasks, runtime initialization etc.

For now, checkpoint restoration is not included in this bucket - the expectation is the next version of the library (v0.0.5) has more definitive recorder and calculator APIs for checkpoint save and restore badput.